app/src/main/java/com/example/mycalculator/MainActivity.kt

```diff
@@ -2,12 +2,88 @@ package com.example.mycalculator
 
 import android.os.Bundle
 import android.support.v7.app.AppCompatActivity
+import com.example.mycalculator.mvp.model.CountModel
+import com.example.mycalculator.mvp.presenter.CountPresenter
+import com.example.mycalculator.mvp.view.CalculatorView
+import com.example.mycalculator.utils.Utils.TEXT_BUTTON_DIV
+import com.example.mycalculator.utils.Utils.TEXT_BUTTON_EIGHT
+import com.example.mycalculator.utils.Utils.TEXT_BUTTON_FIVE
+import com.example.mycalculator.utils.Utils.TEXT_BUTTON_FOUR
+import com.example.mycalculator.utils.Utils.TEXT_BUTTON_MINUS
+import com.example.mycalculator.utils.Utils.TEXT_BUTTON_NINE
+import com.example.mycalculator.utils.Utils.TEXT_BUTTON_ONE
+import com.example.mycalculator.utils.Utils.TEXT_BUTTON_PLUS
+import com.example.mycalculator.utils.Utils.TEXT_BUTTON_POR
+import com.example.mycalculator.utils.Utils.TEXT_BUTTON_SEVEN
+import com.example.mycalculator.utils.Utils.TEXT_BUTTON_SIX
+import com.example.mycalculator.utils.Utils.TEXT_BUTTON_THREE
+import com.example.mycalculator.utils.Utils.TEXT_BUTTON_TWO
+import com.example.mycalculator.utils.Utils.TEXT_BUTTON_ZERO
+import kotlinx.android.synthetic.main.activity_main.*
+
 
 class MainActivity : AppCompatActivity() {
 
+    private var presenter: CountPresenter? = null
```

Comment:
you can use 
lateinit var presenter: CountPresenter
and you can avoid all the safe calls there

Reply:
done!